dataland-keycloak/start_keycloak.sh

```diff
@@ -1,14 +1,29 @@
 #!/bin/bash
 set -eu
 
+mode=$1
+dataland_realm_folder=/opt/keycloak/datalandsecurity
 script_dir="$(dirname "$0")"
 cd $script_dir
 
-if [[ "$1" == initialize ]]; then
+if [[ "$mode" == initialize ]]; then
   echo "Initializing new keycloak realms"
- ./kc.sh import --file /opt/keycloak/data/import/master-realm.json
- ./kc.sh start --import-realm
+  mkdir -p $dataland_realm_folder
+  cp /keycloak_users/datalandsecurity-users-*.json $dataland_realm_folder || echo "No importable users exist"
+  rm $(grep -E -l '"username" : "data_(reader|uploader)"' $dataland_realm_folder/datalandsecurity-users-*.json) || echo "No technical users to be cleaned up"
+  rm $(grep -E -l '"username" : "test_user.*@dataland.com"' $dataland_realm_folder/datalandsecurity-users-*.json) || echo "No test users to be cleaned up"
+  cp /keycloak_realms/datalandsecurity-realm.json $dataland_realm_folder
+  for variable in $(env | grep KEYCLOAK_ | cut -d'=' -f1); do
+    #If one of the env variables containing "KEYCLOAK_" contains a % character the below sed statement will fail and the script will terminate
```

Comment:
Why is this important?

Reply:
The environment variables need to be inserted manually since keycloak only does not do it on its own using the import command.

Reply:
the sed command needs a separator (that can be any character), however the separation character cannot be in the search and replace values since the command crashes otherwise. The usual separator is / however the frontend url does contain / and hence we decided to go with % instead. The comment is intended for other devs to point out that this might be an issue since debugging this would be cumbersome at best.